fdbserver/RocksDBCheckpointUtils.actor.cpp

```diff
@@ -545,6 +545,66 @@ ACTOR Future<Void> RocksDBCheckpointReader::doClose(RocksDBCheckpointReader* sel
 	return Void();
 }
 
+class RocksDBSstFileWriter : public IRocksDBSstFileWriter {
+public:
+	RocksDBSstFileWriter()
+	  : writer(std::make_shared<rocksdb::SstFileWriter>(rocksdb::EnvOptions(), rocksdb::Options())), hasData(false){};
+
+	void open(const std::string localFile) override;
+
+	void write(const KeyRef key, const ValueRef value) override;
+
+	void finish() override;
+
+private:
+	std::shared_ptr<rocksdb::SstFileWriter> writer;
+	std::string localFile;
+	bool hasData;
+};
+
+void RocksDBSstFileWriter::open(const std::string localFile) {
+	rocksdb::Status status;
+	this->localFile = localFile;
+	status = this->writer->Open(this->localFile);
+	if (!status.ok()) {
+		Error e = statusToError(status);
+		TraceEvent(SevError, "RocksDBSstFileWriterWrapperOpenFileError")
+		    .detail("LocalFile", this->localFile)
+		    .detail("Status", status.ToString());
+		throw e;
+	}
+}
+
+void RocksDBSstFileWriter::write(const KeyRef key, const ValueRef value) {
+	rocksdb::Status status;
+	status = this->writer->Put(toSlice(key), toSlice(value));
+	if (!status.ok()) {
+		Error e = statusToError(status);
+		TraceEvent(SevError, "RocksDBSstFileWriterWrapperWriteError")
+		    .detail("LocalFile", this->localFile)
+		    .detail("Key", key)
+		    .detail("Value", value)
+		    .detail("Status", status.ToString());
+		throw e;
+	}
+	this->hasData = true;
+}
+
+void RocksDBSstFileWriter::finish() {
+	if (!this->hasData) {
```

Comment:
Does rocksdb::SstFileWriter open an underlying file? Do we need to call finish() when the nothing is written?

Reply:
finish() throw an error if we call finish() but nothing is written.